RandLAPACK/comps/rl_util.hh

```diff
@@ -370,6 +370,189 @@ void gen_exp_mat(
     }
 }
 
+/// Generates matrix with a staircase spectrum with 4 steps
+template <typename T>
+void gen_step_mat(
+    int64_t& m,
+    int64_t& n,
+    std::vector<T>& A,
+    int64_t k,
+    T cond,
+    bool diagon,
+    RandBLAS::base::RNGState<r123::Philox4x32> state
+) {
```

Comment:
Say what k does, how cond is used, and what ``diagon`` does.

Reply:
Fixed.